### PR TITLE
[Tweak] Default to Traitor instead of Extended when possible

### PR DESCRIPTION
**Pull request by OctoRocket**
## About the PR

This sets the fallback mode to be dependent on player count, while still allowing them be to overridden by an admin. 

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: The game will now fall back to Traitor if there are more than 5 people in the game.

### CODE REVIEW

**Comment by deltanedas** [comment]
imo ideal solution is to keep rerolling until there are no more gamemodes left
so if not enough for zombie, try nukie then try traitor then try extended

**Comment by OctoRocket** [comment]
Considering that this change is primarily for Sala, I don't think it would be best to have super dangerous events all the time. Lowpop zombies and nukies are notorious for being super unfair anyways.

File: Content.Server/GameTicking/GameTicker.GamePreset.cs [hunk 1]
```diff
@@ -43,7 +43,17 @@ void FailedPresetRestart()
...
+                if (fallbackPreset != "")
+                    SetGamePreset(fallbackPreset);
+                else if (startAttempt.Players.Length >= 5)
+                {
+                    SetGamePreset("traitor");
+                }
+                else
+                {
+                    SetGamePreset("extended");
+                }
```
Comment (AJCM-git) [inline Content.Server/GameTicking/GameTicker.GamePreset.cs:56]: This defeats the point of GameLobbyFallbackEnabled and hardcodes gamemodes and timing in the ticker, a much cleaner solution would be making GameLobbyFallbackPreset an array and trying to set the preset in that order, i.e, array is ["traitor", "extended"] so it tries to set traitor and if it fails it tries extended

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/GameTicking/GameTicker.GamePreset.cs (hunk 1 above)
Comment (OctoRocket) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:56]: Where should this array be stored?

Comment (AJCM-git) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:56]: you should be able to just change the cvar def to an array

**Comment by OctoRocket** [comment]
It has been changed to use an array instead.

**Comment by OctoRocket** [comment]
This should be ready for review and merge.

**Comment by AJCM-git** [comment]
Okay apparently cvars dont support collections so this wouldnt work sadly

**Comment by OctoRocket** [comment]
Ah, what would be the optimal solution then?

**Comment by OctoRocket** [comment]
Still probably not the best solution, but it should be easy to switch over if arrays get supported.

**Comment by OctoRocket** [comment]
This PR is ready for review.

File: Content.Server/GameTicking/GameTicker.GamePreset.cs [hunk 2]
```diff
@@ -42,27 +44,38 @@ void FailedPresetRestart()
...
+                    SetGamePreset(preset);
+                    AddGamePresetRules();
+                    StartGamePresetRules();
+
+                    startAttempt.Uncancel();
+                    RaiseLocalEvent(startAttempt);
+
+                    if (!startAttempt.Cancelled)
+                    {
+                        _chatManager.DispatchServerAnnouncement(
```
Comment (deltanedas) [inline Content.Server/GameTicking/GameTicker.GamePreset.cs:62]: this should probably be admin only since it gives players easy meta knowledge

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/GameTicking/GameTicker.GamePreset.cs (hunk 2 above)
Comment (OctoRocket) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs]: Done!

**Comment by OctoRocket** [comment]
It should be ready for a merge now.

**Comment by OctoRocket** [comment]
Well woops

**Comment by OctoRocket** [comment]
Are there any more issues with this PR?

File: Content.Server/GameTicking/GameTicker.GamePreset.cs [hunk 3]
```diff
@@ -17,6 +17,8 @@ namespace Content.Server.GameTicking
     public sealed partial class GameTicker
     {
         public const float PresetFailedCooldownIncrease = 30f;
+        // What it reverts to if it cannot start the fallback, should probably be a cvar or something
+        private const string SecondaryFallback = "Extended";
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/GameTicker.GamePreset.cs:21]: Shouldn't the cvar be an array?

I don't think abraham meant to just hardcode in the secondary.

Comment (OctoRocket) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:21]: Sadly no. I already tried that with this PR and arrays front work with cvars at all.

Comment (metalgearsloth) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:21]: What's wrong with comma separated

Comment (OctoRocket) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:21]: What do you mean?

Comment (OctoRocket) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:21]: Do you mean a string of comma separated values that are then parsed into an array?

Comment (deltanedas) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs:21]: yeah it would be a bit sus but work

Comment (OctoRocket) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs]: It has been done.

Comment (metalgearsloth) [reply Content.Server/GameTicking/GameTicker.GamePreset.cs]: > yeah it would be a bit sus but work

you're sus.

**Pull request merged**